packages/client/src/components/SearchBar/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { IGameFromApi } from "interfaces/api";
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+
+import styles from "./styles.module.scss";
+
+interface SearchBarProps {
+  games: IGameFromApi[];
+}
+
+export const SearchBar: React.FunctionComponent<SearchBarProps> = ({
+  games,
+}) => {
+  const history = useHistory();
+  const [query, setQuery] = useState<string>("");
+  const [isInputActive, setIsInputActive] = useState<boolean>(false);
+  const filteredGames = games.filter((game) =>
+    game.name.toLowerCase().includes(query.toLowerCase())
+  );
```

Comment:
Query search should be implemented on the server's side.
Imagine much more complex search, than just text search.

For example: search across 1000000 items with category + price + rating search.